bindings/cron/cron_test.go

```diff
@@ -64,61 +64,18 @@ func TestCronInitFailure(t *testing.T) {
 
 // TestLongRead
 // go test -v -count=1 -timeout 15s -run TestLongRead ./bindings/cron/.
-func TestCronReadWithDeleteInvoke(t *testing.T) {
+func TestCronRead(t *testing.T) {
 	c := getNewCron()
 	schedule := "@every 1s"
 	assert.NoErrorf(t, c.Init(getTestMetadata(schedule)), "error initializing valid schedule")
-	testsNum := 3
 	i := 0
 	err := c.Read(context.Background(), func(ctx context.Context, res *bindings.ReadResponse) ([]byte, error) {
 		assert.NotNil(t, res)
-		assert.LessOrEqualf(t, i, testsNum, "Invoke didn't stop the schedule")
 		i++
-		if i == testsNum {
-			resp, err := c.Invoke(context.Background(), &bindings.InvokeRequest{
-				Operation: bindings.DeleteOperation,
-			})
-			assert.NoError(t, err)
-			scheduleVal, exists := resp.Metadata["schedule"]
-			assert.Truef(t, exists, "Response metadata doesn't include the expected 'schedule' key")
-			assert.Equal(t, schedule, scheduleVal)
-		}
-
-		return nil, nil
-	})
-	time.Sleep(time.Duration(testsNum+3) * time.Second)
-	assert.Equal(t, testsNum, i)
-	assert.NoErrorf(t, err, "error on read")
-}
-
-func TestCronReadWithContextCancellation(t *testing.T) {
-	c := getNewCron()
-	schedule := "@every 1s"
-	assert.NoErrorf(t, c.Init(getTestMetadata(schedule)), "error initializing valid schedule")
-	testsNum := 3
-	i := 0
-	ctx, cancel := context.WithCancel(context.Background())
-	err := c.Read(ctx, func(ctx context.Context, res *bindings.ReadResponse) ([]byte, error) {
-		assert.NotNil(t, res)
-		assert.LessOrEqualf(t, i, testsNum, "Invoke didn't stop the schedule")
-		i++
-		if i == testsNum {
-			cancel()
-		}
-
 		return nil, nil
 	})
-	time.Sleep(time.Duration(testsNum+3) * time.Second)
-	assert.Equal(t, testsNum, i)
+	time.Sleep(time.Second * 5)
```

Comment:
This test is a place to use the mock clock too  :) Then you can test to make sure it ran exactly 5 times!